packages/legend-extension-dsl-persistence/src/models/metamodels/pure/model/packageableElements/persistence/Persistence.ts

```diff
@@ -0,0 +1,575 @@
+import { PERSISTENCE_HASH_STRUCTURE } from '../../../../../DSLPersistence_ModelUtils';
+import {
+  Class,
+  IdentifiedConnection,
+  PackageableElement,
+  PackageableElementReference,
+  PackageableElementVisitor,
+  Service,
+} from '@finos/legend-graph';
+import { type Hashable, hashArray } from '@finos/legend-shared';
+import { makeObservable, observable, override } from 'mobx';
+
+/**********
+ * persistence
+ **********/
+
+export class Persistence extends PackageableElement implements Hashable {
+  documentation!: string;
+  trigger!: Trigger;
+  service!: PackageableElementReference<Service>;
+  persister!: Persister;
+  notifier!: Notifier;
+
+  constructor(name: string) {
+    super(name);
+
+    makeObservable<Persistence, '_elementHashCode'>(this, {
+      documentation: observable,
+      trigger: observable,
+      service: observable,
+      persister: observable,
+      notifier: observable,
+      _elementHashCode: override,
+    });
+  }
+
+  protected override get _elementHashCode(): string {
+    return hashArray([
+      PERSISTENCE_HASH_STRUCTURE.PERSISTENCE,
+      this.documentation,
+      this.trigger,
+      this.service.hashValue,
+      this.persister,
+      this.notifier,
+    ]);
+  }
+
+  accept_PackageableElementVisitor<T>(
+    visitor: PackageableElementVisitor<T>,
+  ): T {
+    return visitor.visit_PackageableElement(this);
+  }
+}
+
+/**********
+ * trigger
+ **********/
+
+export abstract class Trigger implements Hashable {
+  private readonly _$nominalTypeBrand!: 'Trigger';
```

Comment:
I think you don't need `nomialTypeBrand` anymore, we plan to remove them in the future across the codebase.

Reply:
Fixed in #990